diff --git a/src/backend/catalog/genbki.sh b/src/backend/catalog/genbki.sh
index c6b0cf0..035449d 100644
--- a/src/backend/catalog/genbki.sh
+++ b/src/backend/catalog/genbki.sh
@@ -77,7 +77,7 @@ sed -e "s/;[ 	]*$//g" \
     -e "s/[ 	]Oid/\ oid/g" \
     -e "s/[ 	]NameData/\ name/g" \
     -e "s/^Oid/oid/g" \
-    -e "s/^NameData/\name/g" \
+    -e "s/^NameData/name/g" \
     -e "s/(NameData/(name/g" \
     -e "s/(Oid/(oid/g" \
 	-e "s/NAMEDATALEN/$NAMEDATALEN/g" \
diff --git a/src/backend/commands/Makefile b/src/backend/commands/Makefile
index 7e4fe41..a348e94 100644
--- a/src/backend/commands/Makefile
+++ b/src/backend/commands/Makefile
@@ -14,7 +14,7 @@ include ../../Makefile.global
 CFLAGS += -I..
 
 OBJS = async.o creatinh.o command.o copy.o defind.o define.o \
-       remove.o rename.o vacuum.o version.o view.o cluster.o \
+       remove.o rename.o vacuum.o  view.o cluster.o \
        recipe.o explain.o sequence.o trigger.o user.o proclang.o \
        dbcommands.o variable.o
 
diff --git a/src/backend/libpq/pqcomm.c b/src/backend/libpq/pqcomm.c
index 41280c8..2a65de1 100644
--- a/src/backend/libpq/pqcomm.c
+++ b/src/backend/libpq/pqcomm.c
@@ -532,7 +532,6 @@ pq_async_notify()
 	}
 	else
 	{
-		extern int	errno;
 
 		printf("SIGURG but no data: len = %d, err=%d\n", len, errno);
 	}
@@ -735,7 +734,6 @@ StreamOpen(char *hostName, short portName, Port *port)
 	int			len,
 				err;
 	struct hostent *hp;
-	extern int	errno;
 
 	/* set up the server (remote) address */
 	MemSet((char *) &port->raddr, 0, sizeof(port->raddr));
diff --git a/src/backend/storage/ipc/ipc.c b/src/backend/storage/ipc/ipc.c
index 28fbd87..203c454 100644
--- a/src/backend/storage/ipc/ipc.c
+++ b/src/backend/storage/ipc/ipc.c
@@ -383,7 +383,6 @@ static int	IpcSemaphoreLock_return;
 void
 IpcSemaphoreLock(IpcSemaphoreId semId, int sem, int lock)
 {
-	extern int	errno;
 	int			errStatus;
 	struct sembuf sops;
 
@@ -427,7 +426,6 @@ static int	IpcSemaphoreUnlock_return;
 void
 IpcSemaphoreUnlock(IpcSemaphoreId semId, int sem, int lock)
 {
-	extern int	errno;
 	int			errStatus;
 	struct sembuf sops;
 
diff --git a/src/backend/utils/adt/float.c b/src/backend/utils/adt/float.c
index 6f27e14..90d00f6 100644
--- a/src/backend/utils/adt/float.c
+++ b/src/backend/utils/adt/float.c
@@ -105,7 +105,6 @@ extern double rint(double x);
 
 #endif
 
-extern int	isinf(double x);
 
 #endif
 /* ========== USER I/O ROUTINES ========== */
diff --git a/src/backend/utils/error/elog.c b/src/backend/utils/error/elog.c
index 63fd69b..3c60218 100644
--- a/src/backend/utils/error/elog.c
+++ b/src/backend/utils/error/elog.c
@@ -47,8 +47,6 @@ elog(int lev, const char *fmt,...)
 				line[ELOG_MAXLEN];
 	char	   *bp;
 	const char *cp;
-	extern int	errno,
-				sys_nerr;
 
 #ifndef PG_STANDALONE
 	extern FILE *Pfout;
diff --git a/src/backend/utils/error/exc.c b/src/backend/utils/error/exc.c
index ee79fa9..243cec5 100644
--- a/src/backend/utils/error/exc.c
+++ b/src/backend/utils/error/exc.c
@@ -103,8 +103,6 @@ ExcPrint(Exception *excP,
 		 ExcData data,
 		 ExcMessage message)
 {
-	extern int	errno;
-	extern int	sys_nerr;
 
 #ifdef	lint
 	data = data;
diff --git a/src/backend/utils/sort/psort.c b/src/backend/utils/sort/psort.c
index fa439f5..91b82de 100644
--- a/src/backend/utils/sort/psort.c
+++ b/src/backend/utils/sort/psort.c
@@ -19,6 +19,9 @@
  * file. You must now explicitly call an interface function to end the sort,
  * psort_end, when you are done.
  *		Now most of the global variables are stuck in the Sort nodes, and
+#include <errno.h>
+#include <errno.h>
+#include <errno.h>
  * accessed from there (they are passed to all the psort routines) so that
  * each sort running has its own separate state. This is facilitated by having
  * the Sort nodes passed in to all the interface functions.
@@ -33,6 +36,7 @@
  *				MAXMERGE, MAXTAPES
  *-------------------------------------------------------------------------
  */
+#include <errno.h>
 #include <stdio.h>
 #include <math.h>
 #include <string.h>
@@ -1018,7 +1022,6 @@ gettape()
 	static int	tapeinit = 0;
 	char	   *mktemp();
 	static unsigned int uniqueFileId = 0;
-	extern int	errno;
 	char		uniqueName[MAXPGPATH];
 
 	tp = (struct tapelst *) palloc((unsigned) sizeof(struct tapelst));
diff --git a/src/bin/psql/stringutils.c b/src/bin/psql/stringutils.c
index 62d8fef..a02ab1e 100644
--- a/src/bin/psql/stringutils.c
+++ b/src/bin/psql/stringutils.c
@@ -17,7 +17,7 @@
 #include <stdlib.h>
 
 #include "postgres.h"
-#ifndef HAVE_STRDUP
+#if 0
 #include "strdup.h"
 #endif
 
diff --git a/src/include/c.h b/src/include/c.h
index bbc8426..730e078 100644
--- a/src/include/c.h
+++ b/src/include/c.h
@@ -44,13 +44,16 @@
 #define C_H
 
 /* We have to include stdlib.h here because it defines many of these macros
+#include <errno.h>
    on some platforms, and we only want our definitions used if stdlib.h doesn't
    have its own.
 */
 #include <stdlib.h>
+#include <errno.h>
 
 /* ----------------------------------------------------------------
  *				Section 1:	bool, true, false, TRUE, FALSE
+#include <errno.h>
  * ----------------------------------------------------------------
  */
 /*
diff --git a/src/include/postgres.h b/src/include/postgres.h
index 12c818a..4286b94 100644
--- a/src/include/postgres.h
+++ b/src/include/postgres.h
@@ -175,9 +175,9 @@ typedef uint32 CommandId;
 #define CATALOG(x) \
 	typedef struct CppConcat(FormData_,x)
 
-#define DATA(x) extern int errno
-#define DESCR(x) extern int errno
-#define DECLARE_INDEX(x) extern int errno
+#define DATA(x) extern int no_such_variable
+#define DESCR(x) extern int no_such_variable
+#define DECLARE_INDEX(x) extern int no_such_variable
 
 #define BUILD_INDICES
 #define BOOTSTRAP
diff --git a/src/interfaces/ecpg/preproc/extern.h b/src/interfaces/ecpg/preproc/extern.h
index 53da42c..66b80b5 100644
--- a/src/interfaces/ecpg/preproc/extern.h
+++ b/src/interfaces/ecpg/preproc/extern.h
@@ -3,8 +3,8 @@
 extern int	debugging,
 			braces_open;
 extern char *yytext;
-extern int	yylineno,
-			yyleng;
+extern int yylineno;
+extern size_t yyleng;
 extern FILE *yyin,
 		   *yyout;
 
