--- ./src/interfaces/ecpg/preproc/preproc.y	2016-02-14 00:24:33.861223994 +0300
+++ ./src/interfaces/ecpg/preproc/preproc.y.hacked	2016-02-14 00:24:33.933225737 +0300
@@ -790,7 +790,7 @@
 %type  <str>	opt_table opt_union opt_unique sort_clause sortby_list
 %type  <str>	sortby OptUseOp opt_inh_star relation_name_list name_list
 %type  <str>	group_clause having_clause from_clause c_list 
-%type  <str>	table_list join_outer where_clause relation_expr row_op sub_type
+%type  <str> join_outer where_clause relation_expr row_op sub_type
 %type  <str>	opt_column_list insert_rest InsertStmt OptimizableStmt
 %type  <str>    columnList DeleteStmt LockStmt UpdateStmt CursorStmt
 %type  <str>    NotifyStmt columnElem copy_dirn c_expr UnlistenStmt
@@ -824,7 +824,7 @@
 %type  <str>	case_expr when_clause_list case_default case_arg when_clause
 %type  <str>    select_clause opt_select_limit select_limit_value
 %type  <str>    select_offset_value table_list using_expr join_expr
-%type  <str>	using_list from_expr table_expr join_clause join_type
+%type  <str>	using_list from_expr table_expr join_type
 %type  <str>	join_qual update_list join_clause join_clause_with_union
 %type  <str>	opt_level opt_lock lock_type
 
