diff --git a/src/bin/psql/tab-complete.c b/src/bin/psql/tab-complete.c
index f8c1919..2e2cebd 100644
--- a/src/bin/psql/tab-complete.c
+++ b/src/bin/psql/tab-complete.c
@@ -719,7 +719,7 @@ psql_completion(char *text, int start, int end)
 		   strcmp(prev_wd, "\\w") == 0 || strcmp(prev_wd, "\\write") == 0
 		)
 	{
-		matches = completion_matches(text, filename_completion_function);
+		matches = completion_matches(text, rl_filename_completion_function);
 	}
 
 
diff --git a/src/bin/psql/tab-complete.c b/src/bin/psql/tab-complete.c
index 0e9cb2d..7d8cf47 100644
--- a/src/bin/psql/tab-complete.c
+++ b/src/bin/psql/tab-complete.c
@@ -700,7 +700,7 @@ psql_completion(char *text, int start, int end)
 		)
 	{
 #ifdef HAVE_FILENAME_COMPLETION_FUNCTION
-		matches = completion_matches(text, filename_completion_function);
+		matches = completion_matches(text, rl_filename_completion_function);
 #else
 
 		/*
--- src/bin/psql/tab-complete.c
+++ src/bin/psql/tab-complete.c
@@ -719,7 +719,7 @@ psql_completion(char *text, int start, int end)
 		   strcmp(prev_wd, "\\w") == 0 || strcmp(prev_wd, "\\write") == 0
 		)
 	{
-		matches = completion_matches(text, filename_completion_function);
+		matches = completion_matches(text, rl_filename_completion_function);
 	}
 
 
