diff --git a/src/include/parser/gramparse.h b/src/include/parser/gramparse.h
index 4b061e0..36edd9a 100644
--- a/src/include/parser/gramparse.h
+++ b/src/include/parser/gramparse.h
@@ -33,6 +33,11 @@
 #define YYLTYPE  int
 
 /*
+ * The type of yyscanner is opaque outside scan.l.
+ */
+typedef void *base_yyscan_t;
+
+/*
  * After defining YYLTYPE, it's safe to include gram.h.
  */
 #include "parser/gram.h"
@@ -94,10 +99,6 @@ typedef struct base_yy_extra_type
 	bool		QueryIsRule;	/* signals we are parsing CREATE RULE */
 } base_yy_extra_type;
 
-/*
- * The type of yyscanner is opaque outside scan.l.
- */
-typedef void *base_yyscan_t;
 
 /*
  * In principle we should use yyget_extra() to fetch the yyextra field
