diff --git a/configure b/configure
index 940fc6a..bd6a38e 100755
--- a/configure
+++ b/configure
@@ -2763,7 +2763,7 @@ else
       then
         echo '%%'  > conftest.l
         if $pgac_candidate -t conftest.l 2>/dev/null | grep FLEX_SCANNER >/dev/null 2>&1; then
-          if $pgac_candidate --version | grep '2\.5\.3' >/dev/null 2>&1; then
+          if $pgac_candidate --version | grep '2\.5\.3$' >/dev/null 2>&1; then
             pgac_broken_flex=$pgac_candidate
             continue
           fi
