diff --git a/src/backend/commands/dbcommands.c b/src/backend/commands/dbcommands.c
index bda6484..6e18401 100644
--- a/src/backend/commands/dbcommands.c
+++ b/src/backend/commands/dbcommands.c
@@ -144,20 +144,12 @@ createdb(const char *dbname, const char *dbpath, int encoding)
 			 DataDir, SEP_CHAR, SEP_CHAR, SEP_CHAR, loc);
 	ret = system(buf);
 	/* Some versions of SunOS seem to return ECHILD after a system() call */
-#if defined(sun)
 	if (ret != 0 && errno != ECHILD)
-#else
-	if (ret != 0)
-#endif
 	{
 		/* Failed, so try to clean up the created directory ... */
 		snprintf(buf, sizeof(buf), "rm -rf '%s'", loc);
 		ret = system(buf);
-#if defined(sun)
 		if (ret == 0 || errno == ECHILD)
-#else
-		if (ret == 0)
-#endif
 			elog(ERROR, "CREATE DATABASE: could not initialize database directory");
 		else
 			elog(ERROR, "CREATE DATABASE: Could not initialize database directory. Delete failed as well");
