diff --git a/src/interfaces/ecpg/preproc/extern.h b/src/interfaces/ecpg/preproc/extern.h
index 06cc74f..cd1cda3 100644
--- a/src/interfaces/ecpg/preproc/extern.h
+++ b/src/interfaces/ecpg/preproc/extern.h
@@ -30,8 +30,8 @@ extern char *yytext,
 #ifdef YYDEBUG
 extern int	yydebug;
 #endif
-extern int	yylineno,
-			yyleng;
+extern int yylineno;
+extern size_t yyleng;
 extern FILE *yyin,
 		   *yyout;
 
diff --git a/src/interfaces/ecpg/preproc/extern.h b/src/interfaces/ecpg/preproc/extern.h
index 5628e30..5ad4e67 100644
--- a/src/interfaces/ecpg/preproc/extern.h
+++ b/src/interfaces/ecpg/preproc/extern.h
@@ -19,8 +19,8 @@ extern char *connection;
 extern char *input_filename;
 extern char *yytext,
 			errortext[128];
-extern int	yylineno,
-			yyleng;
+extern int	yylineno;
+extern size_t yyleng;
 extern FILE *yyin,
 		   *yyout;
 
